internal/controller/nnf_workflow_controller_helpers.go

```diff
@@ -713,37 +713,86 @@ func (r *NnfWorkflowReconciler) getLustreMgsFromPool(ctx context.Context, pool s
 		return corev1.ObjectReference{}, "", dwsv1alpha2.NewResourceError("").WithUserMessage("no MGSs found for pool: %s", pool).WithFatal().WithUser()
 	}
 
-	persistentStorage := persistentStorageList.Items[rand.Intn(len(persistentStorageList.Items))]
+	healthyMgts := make(map[string]corev1.ObjectReference)
+	for _, persistentStorage := range persistentStorageList.Items {
+		// Find the NnfStorage for the PersistentStorage so we can check it's status and get the MGT LNid
+		nnfStorage := &nnfv1alpha1.NnfStorage{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      persistentStorage.Name,
+				Namespace: persistentStorage.Namespace,
+			},
+		}
 
-	// Find the NnfStorage for the PersistentStorage so we can get the LNid
-	nnfStorage := &nnfv1alpha1.NnfStorage{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      persistentStorage.Name,
-			Namespace: persistentStorage.Namespace,
-		},
-	}
+		if err := r.Get(ctx, client.ObjectKeyFromObject(nnfStorage), nnfStorage); err != nil {
+			return corev1.ObjectReference{}, "", dwsv1alpha2.NewResourceError("could not get persistent NnfStorage %v for MGS", client.ObjectKeyFromObject(nnfStorage)).WithError(err)
```

Comment:
Log it, and go to the next one.

Reply:
I'd rather not do that in case it's the only MGT that's available. If the cache is stale we might get this error, but it's marked as a Minor() error (the default), so it never even makes it out to flux anyway. It just requeues the reconciler to try again.